### ISSUE TITLE
Dependency Alert - Razor Language Services 

### ISSUE THREAD

**Issue by RasicN** [aspnet/Blazor#339]
When trying to download the extension through Visual Studio I get an error that Razor Language Services must be installed, though as you can see it is!

![image](https://user-images.githubusercontent.com/21132694/37840995-8884c7c2-2e8c-11e8-8966-5114c32b8863.png)

**Comment by rynowak** [comment]
Hi,

Can you please confirm that this is on VS **15.7 preview 1**  with the **ASP NET and web development** workload installed. 

![image](https://user-images.githubusercontent.com/1430011/37841931-18a948d2-2e7e-11e8-8a27-b5936128e381.png)

You can see this by opening the Visual Studio Installer and clicking on Modify

See also: https://github.com/aspnet/Blazor/issues/328

**Comment by RasicN** [comment]
I followed this step on getting started:

- Install the latest preview of Visual Studio 2017 (15.7) with the Web development workload.

Which appears to be 15.7.0 preview 2.0 and did got the "Asp.Net and Web Development" module which is what I assumed **Web development workload** meant.  I will try and get .Net Desktop Development but we may want to update the README to be more specific if that is the issue.

**Comment by RasicN** [comment]
Still not working :(

![image](https://user-images.githubusercontent.com/21132694/37843297-869bd60c-2e92-11e8-95e7-963e0d5c2f64.png)

**Comment by rynowak** [comment]
When you try to install now is the failure the same? 

I'm taking a look at rebuilding this with different metadata, you're the second person to report a problem installing.

**Comment by RasicN** [comment]
Yep same problem

**Comment by rynowak** [comment]
OK thanks for the report on this. I'm rebuilding the VSIX to have a better set of dependencies. In the mean time could you do a repair of your VS install, and then I'll notify you when the new build is published.

**Comment by rynowak** [comment]
@RasicN - 15.7.10142 we've uploaded a new version and clarified the instructions https://marketplace.visualstudio.com/items?itemName=aspnet.blazor 

Let us know if this fixes the issue!

**Comment by RasicN** [comment]
It has made it past the Razor Language Services requirement but now I am getting an error when it tries to install .Net.CoreSDK...

`3/23/2018 6:32:17 PM - Install Error : Microsoft.VisualStudio.Setup.PackageFailureException: Package 'Microsoft.Net.Core.SDK' failed to install
   at Microsoft.VisualStudio.Setup.InstallOperation.Run(CancellationToken token)
   at Microsoft.VisualStudio.Setup.Engine.RunOperation(InstallOperation installOperation, CancellationToken token, ExecuteAction action, ITelemetryOperation telemetryOperation)
   at Microsoft.VisualStudio.Setup.Engine.RunCoreOperation(InstallOperation coreOperation, ExecuteAction action, ITelemetryOperation telemetryOperation, CancellationToken token)
   at Microsoft.VisualStudio.Setup.Engine.Install(Product product, String destination, CancellationToken token)
   at Microsoft.VisualStudio.ExtensionManager.ExtensionEngineImpl.PerformSetupEngineInstall(InstallableExtensionImpl extension, Boolean installPerMachine, Boolean isPackComponent, IDictionary`2 extensionsInstalledSoFar, List`1 extensionsUninstalledSoFar, IInstalledExtensionList modifiedInstalledExtensionsList, IProgress`1 progress, InstallFlags installFlags, AsyncOperation asyncOp, Version targetedVsVersion, IInstalledExtension& newExtension)
   at Microsoft.VisualStudio.ExtensionManager.ExtensionEngineImpl.InstallInternal(InstallableExtensionImpl extension, InstallFlags installFlags, IDictionary`2 extensionsInstalledSoFar, List`1 extensionsUninstalledSoFar, IInstalledExtensionList modifiedInstalledExtensionsList, AsyncOperation asyncOp, IProgress`1 progress, Version targetedVsVersion)
   at Microsoft.VisualStudio.ExtensionManager.ExtensionEngineImpl.BeginInstall(IInstallableExtension installableExtension, InstallFlags installFlags, AsyncOperation asyncOp, Version targetedVsVersion)
   at Microsoft.VisualStudio.ExtensionManager.ExtensionEngineImpl.InstallWorker(IInstallableExtension extension, InstallFlags installFlags, AsyncOperation asyncOp)`

Going to try and re-install .Net Core SDK

**Comment by RasicN** [comment]
![image](https://user-images.githubusercontent.com/21132694/37857463-366174b2-2ec9-11e8-9023-3cb89fadbc0a.png)

Is this the correct SDK version?

**Comment by jt2** [comment]
It works for me with that version.

I had the same problem you had with "Razor Language Services", I did a repair on the preview from the installer and it start working.

**Comment by RasicN** [comment]
Will give it a go.

**Comment by rynowak** [comment]
Yes, if you installed "Razor Language Services" from the VS gallery, you will need to repair. That extension is for a a very old version of VS and will break newer versions.

**Comment by RasicN** [comment]
Same thing after the repair.  I will try from scratch tomorrow on a different PC.  It could be the current machine I'm on.  Windows 7 shouldn't matter should it?

**Comment by rynowak** [comment]
Thanks for your patience with this @RasicN 

If you don't mind, try to capture an exact step by step of what you're doing when you install. That will help in case others run into the same kind of problem.

**Comment by rynowak** [comment]
Closing this because I think we've resolved all of the issues with installing the extension

**Comment by redar84** [comment]
Guys I bought a new Laptop and I installed VS2017 Enterprise edition provided by my work. I tried to install the Blazor but I am getting erorr. I installed the .net core 2.2 and run VS2017. On my surface Blazor is running on 2.1 core. I attached the error I a getting here
![Capture](https://user-images.githubusercontent.com/10550117/54871958-8f820800-4d8a-11e9-8157-8806f2c6c526.PNG)
. Thank you for your help
